### DISCUSSION TITLE
Is USB HID possible?

### DISCUSSION THREAD

**Discussion by macron0** [dbuezas/lgt8fx#103] [Q&A]
Hi 

I am planing to try and make a Amiga/C64/Atari joystick adapter. I glimsped at some items a about it a long time ago. But It seems you need special kind of atmega chip to do this. (32U4) Which is on the pro micro.
I think they just emulate a keyboard and than map the keys to Winuae. 

So before I waste a weekend on something that is not possible does any one know if this would be possible with the lgt8fx? 

I have the green nano look alike and the smaller SSOP20 version.

**Comment by jg1uaa** [comment]
ATmega32U4 series has native USB interface but LGT8Fx does not.
LGT8Fx board support package has V-USB (Virtual USB) at libraries/usbdrv but the code is still for original AVR.
If we want to use V-USB on LGT8Fx maybe we have to recalculate instruction-execution-timing with hard effort...
(V-USB has include files to support 12, 12.8, 15, 16, 16.5, 18 and 20MHz respectively.)

**Comment by LaZsolt** [comment]
Massduino UNO Pro R3 (with LGT8F328-LQFP48) have V-USB capabilities. Check this pages and try they Massduino core:
http://www.inhaos.com/product_info.php?products_id=168
http://www.inhaos.com/download.php

**Comment by LaZsolt** [comment]
https://github.com/dimag0g/massduino_boards

**Comment by prosper00** [comment]
I made a joystick converter earlier this year - https://github.com/prosper00/GRiP-duino  At the time, I chose to go with a pro micro. I thought about implementing with VUSB, however, bit banging USB is quite sensitive and cpu-intensive, and I didn't want to have to worry about latency issues, or missing interrupts/dropping data. In retrospect, it probably wouldn't have been an issue.

Also in retrospect, I would have built it on an STM32 - the cheapest uc I can think of that supports native USB. Less than half the price of a (clone) pro micro.

**Comment by macron0** [comment]
Hi all. 
Thank you for your replies.  
I ordered a pro micro so I will try that one first (when it arrives).
Messing with the timings and the usb protocol itself is way above my skill level. I do have an stm32 . Maybe I try that too.  
It would have been cool if the LGT8F328 board could manage it since I have a bunch of those. 
I did not know about massduino. But the price is about the same as  a pro micro. So I gues for this project I stick to that.
